client.go

```diff
@@ -153,8 +153,22 @@ func (c *Client) Ls(ctx context.Context, opts ...LsOption) (chan PinStatusGetter
 	}
 
 	go func() {
-		defer close(errs)
-		defer close(res)
+		defer func() {
+			if r := recover(); r != nil {
+				var err error
+				switch x := r.(type) {
+				case string:
+					err = fmt.Errorf("unexpected error while listing remote pins: %s", x)
+				case error:
+					err = fmt.Errorf("unexpected error while listing remote pins: %w", x)
+				default:
+					err = errors.New("unknown panic while listing remote pins")
+				}
+				errs <- err
+			}
+			close(errs)
+			close(res)
+		}()
```

Comment:
Is this ok or overkill?

Reply:
SGTM

Maybe we should wrap this in a wrapper package ? Like the `multierror.Group`.